front/src/pages/settings/data-model/SettingsObjectDetail.tsx

```diff
@@ -33,49 +34,51 @@ export const SettingsObjectDetail = () => {
   const navigate = useNavigate();
 
   const { objectSlug = '' } = useParams();
-  const { activeObjects, disableObject, findActiveObjectBySlug } =
+  const { disableObject, findActiveObjectBySlug, loading } =
     useObjectMetadata();
   const activeObject = findActiveObjectBySlug(objectSlug);
 
   useEffect(() => {
-    if (activeObjects.length && !activeObject) {
-      navigate(AppPath.NotFound);
-    }
-  }, [activeObject, activeObjects.length, navigate]);
+    if (loading) return;
+    if (!activeObject) navigate(AppPath.NotFound);
+  }, [activeObject, loading, navigate]);
 
   const { activateField, disableField, eraseField } = useFieldMetadata();
-  const activeFields = activeObject?.fields.filter(
+
+  if (!activeObject) return null;
+
+  const activeFields = activeObject.fields.filter(
     (fieldItem) => fieldItem.isActive,
   );
-  const disabledFields = activeObject?.fields.filter(
+  const disabledFields = activeObject.fields.filter(
     (fieldItem) => !fieldItem.isActive,
   );
 
+  const handleDisable = async () => {
+    await disableObject(activeObject);
+    navigate('/settings/objects');
+  };
+
   return (
     <SubMenuTopBarContainer Icon={IconSettings} title="Settings">
       <SettingsPageContainer>
         <Breadcrumb
           links={[
             { children: 'Objects', href: '/settings/objects' },
-            { children: activeObject?.labelPlural ?? '' },
+            { children: activeObject.labelPlural },
           ]}
         />
-        {activeObject && (
-          <SettingsAboutSection
-            iconKey={activeObject.icon ?? undefined}
-            name={activeObject.labelPlural || ''}
-            isCustom={activeObject.isCustom}
-            onDisable={() => {
-              disableObject(activeObject);
-              navigate('/settings/objects');
-            }}
-            onEdit={() => navigate('./edit')}
-          />
-        )}
+        <SettingsAboutSection
+          iconKey={activeObject.icon ?? undefined}
+          name={activeObject.labelPlural || ''}
+          isCustom={activeObject.isCustom}
+          onDisable={handleDisable}
+          onEdit={() => navigate('./edit')}
+        />
```

Comment:
Think we should not display 'edit' for non editable Standard Objects
<img width="1348" alt="image" src="https://github.com/twentyhq/twenty/assets/29927851/41de4738-a724-4b86-9255-be95d223a09a">

Reply:
According to the original issue the Standard objects also have the "Edit" option: https://github.com/twentyhq/twenty/issues/1765 (see "Edit an object" section).
I'd suggest you raise this concern with @Bonapara :)